lib/chef/policy_builder/policyfile.rb

```diff
@@ -356,6 +354,30 @@ def config
         Chef::Config
       end
 
+      private
+
+      def compat_mode_manifest_for(cookbook_name, lock_data)
```

Comment:
Yeah, I noticed but I decided to leave it because in a future change we'll be using `lock_data["identifier"]` instead of `lock_data["dotted_decimal_identifier"]` so the code paths will diverge more. And compete mode will eventually get completely removed.